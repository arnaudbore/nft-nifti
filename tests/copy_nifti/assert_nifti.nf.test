nextflow_process {

  name "Test COPY_NIFTI"

  script "tests/copy_nifti/copy_nifti.nf"
  process "COPY_NIFTI"

  test("Test - nft-nifti - precision 2") {
    when {
      process {
        """
        input[0] = file("$baseDir/tests/dummy.nii.gz", checkIfExists: true)
        """
      }
    }
    then {
      assert snapshot(niftiMD5SUM(process.out.nifti.get(0), 2)).match()
    }
  }

  test("Test - nft-nifti - precision default") {
    when {
      process {
        """
        input[0] = file("$baseDir/tests/dummy.nii.gz", checkIfExists: true)
        """
      }
    }
    then {
      assert snapshot(niftiMD5SUM(process.out.nifti.get(0))).match()
    }
  }

  test("Test - nft-nifti - precision 8") {
    when {
      process {
        """
        input[0] = file("$baseDir/tests/dummy.nii.gz", checkIfExists: true)
        """
      }
    }
    then {
      assert snapshot(niftiMD5SUM(process.out.nifti.get(0), 8)).match()
    }
  }
}