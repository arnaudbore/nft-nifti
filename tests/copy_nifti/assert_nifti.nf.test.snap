{
    "Test - nft-nifti - precision 2": {
        "content": [
            "output.nii.gz:md5:header,a1d6f9fc0fa0c648e2b761a807007a61,data,505e762062a1916f7c405c913c938585"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-30T12:08:01.055498814"
    },
    "Test - nft-nifti - precision 8": {
        "content": [
            "output.nii.gz:md5:header,a1d6f9fc0fa0c648e2b761a807007a61,data,6529e4a8556b1800e6da5f973868c447"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-30T12:08:12.521603918"
    },
    "Test - nft-nifti - precision default": {
        "content": [
            "output.nii.gz:md5:header,a1d6f9fc0fa0c648e2b761a807007a61,data,6529e4a8556b1800e6da5f973868c447"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-30T12:08:06.432466035"
    }
}